plugins/inventory/digitalocean.py

```diff
@@ -0,0 +1,215 @@
+      - This value cannot be greater than 200.
+    type: int
+    default: 200
+'''
+
+EXAMPLES = r'''
+# Using keyed groups and compose for hostvars
+plugin: community.digitalocean.digitalocean
+api_token: "{{ api_token }}"
+attributes:
+  - id
+  - name
+  - memory
+  - vcpus
+  - disk
+  - size
+  - image
+  - networks
+  - volume_ids
+  - tags
+  - region
+keyed_groups:
+  - key: do_region.slug
+    prefix: 'region'
+    separator: '_'
+  - key: do_tags | lower
+    prefix: ''
+    separator: ''
+compose:
+  ansible_host: do_networks.v4 | selectattr('type','eq','public')
+    | map(attribute='ip_address') | first
+  class: do_size.description | lower
+  distro: do_image.distribution | lower
+'''
+
+import re
+import json
+from ansible.errors import AnsibleParserError
+from ansible.module_utils.urls import Request
+from ansible.module_utils.six.moves.urllib.error import URLError, HTTPError
+from ansible.plugins.inventory import BaseInventoryPlugin, Constructable, Cacheable
+
+
+class InventoryModule(BaseInventoryPlugin, Constructable, Cacheable):
+
+    NAME = 'community.digitalocean.digitalocean'
+
+    def _validate_config(self, config):
+        if 'var_prefix' in config:
+            if not re.match('^[a-z][a-z_]*$', config['var_prefix']):
+                raise AnsibleParserError("var_prefix contains invalid characters")
+
+        return True
+
+    def verify_file(self, path):
+        valid = False
+        if super(InventoryModule, self).verify_file(path):
+            if path.endswith(
+                    ('do_hosts.yaml', 'do_hosts.yml',
+                     'digitalocean.yaml', 'digitalocean.yml',
+                     'digital_ocean.yaml', 'digital_ocean.yml')
+            ):
+                valid = True
+            else:
+                self.display.vvv(
+                    'Skipping due to inventory source file name mismatch. '
+                    'The file name has to end with one of the following: '
+                    'do_hosts.yaml, do_hosts.yml '
+                    'digitalocean.yaml, digitalocean.yml, '
+                    'digital_ocean.yaml, digital_ocean.yml.')
+        return valid
+
+    def _get_payload(self):
+        # request parameters
+        api_token = self.get_option('api_token')
+        headers = {
+            'Content-Type': 'application/json',
+            'Authorization': 'Bearer {0}'.format(api_token)
+        }
+
+        # build url
+        pagination = self.get_option('pagination')
+        url = 'https://api.digitalocean.com/v2/droplets?per_page=' + str(pagination)
+
+        # send request(s)
+        self.req = Request(headers=headers)
+        payload = []
+        try:
+            while url:
+                self.display.vvv('Sending request to {0}'.format(url))
+                response = json.load(self.req.get(url))
+                payload.extend(response['droplets'])
+                url = response.get('links', {}).get('pages', {}).get('next')
```

Comment:
Did someone test whether that works (for example by setting pagination to 1, or whatever is the smallest value DO allows)?

Reply:
I tested it with different values between 2 and 25 and worked as expected.